pqarrow/arrow.go

```diff
@@ -796,21 +796,19 @@ func (c *ParquetConverter) writeColumnToArray(
 				return fmt.Errorf("write page: %w", err)
 			}
 		default:
-			if n := p.NumValues(); int64(cap(c.scratchValues)) < n {
+			if n := columnChunk.NumValues(); int64(cap(c.scratchValues)) < n {
 				c.scratchValues = make([]parquet.Value, n)
 			} else {
 				c.scratchValues = c.scratchValues[:n]
 			}
-
-			// We're reading all values in the page so we always expect an io.EOF.
 			reader := p.Values()
-			if _, err := reader.ReadValues(c.scratchValues); err != nil && err != io.EOF {
+			i, err = reader.ReadValues(c.scratchValues[i:])
```

Comment:
Do we want to limit the buffering somehow?

Reply:
We didn't before but something to consider in the future potentiall

Reply:
But there was no limit to how many values were in the page just like with row groups so it was a virtual limit not one that we set.